common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -95,6 +95,20 @@ case class CompletionMessage(override val transid: TransactionId,
 }
 
 object CompletionMessage extends DefaultJsonProtocol {
+  implicit def eitherResponse =
+    new JsonFormat[Either[ActivationId, WhiskActivation]] {
+      def write(either: Either[ActivationId, WhiskActivation]) = either match {
+        case Right(a) => a.toJson
+        case Left(b)  => b.toJson
+      }
+
+      def read(value: JsValue) = value match {
+        case _: JsString => Left(value.convertTo[ActivationId])
```

Comment:
Can you add a comment that we guarantee activation id to be a string even if it’s contains only digits?

Reply:
Done!